### ISSUE TITLE
Azure DevOps Causes a Differing Behavior (Start-Process)

### ISSUE THREAD

**Issue by Silvenga** [PowerShell/PowerShell#11405]
I was asked by the Azure DevOps support team (case `119110624002446`) to open this bug as it was believed that this is a bug in PowerShell Core, and not Azure DevOps.

# Steps to reproduce

Given a `azure-pipeline.yaml`

```yaml
jobs:
- job: HelloWorld
  pool:
    vmImage: ubuntu-latest
  steps:
  - powershell: |
      Start-Process "tail" -Args "-f /dev/null" 
      exit 0
    timeoutInMinutes: 1
```

# Expected behavior

The task should not timeout.

When executed from a local Ubuntu machine:

```
~$ pwsh
PS> Start-Process "tail" -Args "-f /dev/null"
PS> ps
   PID TTY          TIME CMD
31187 pts/0    00:00:00 bash
31236 pts/0    00:00:02 pwsh
31398 pts/0    00:00:00 tail
31404 pts/0    00:00:00 ps
PS> exit
~$ ps
   PID TTY          TIME CMD
31187 pts/0    00:00:00 bash
31398 pts/0    00:00:00 tail
31423 pts/0    00:00:00 ps
```

The main process (pwsh) does complete and the tail process's parent becomes init:

```
~$ pstree -s -p 31398
systemd(1)───tail(31398)
```

Which is desired behavior.

# Actual behavior

The script exists with exit code 0, and then the task times out waiting for the pwsh process to stop (assumed).

# Environment data

```none
Name                           Value
----                           -----
PSVersion                      6.2.3
PSEdition                      Core
GitCommitId                    6.2.3
OS                             Linux 4.15.0-1064-azure #69-Ubuntu SMP Tue Nov …
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by iSazonov** [comment]
@SteveL-MSFT Please triage the issue.

**Comment by Silvenga** [comment]
Also, a separate case `120022124006034` for PowerShell running under Azure DevOps is suggesting that this issue impacts **PowerShell 5** on Windows Server.

@iSazonov or @SteveL-MSFT can you provide any insights into this?

**Comment by iSazonov** [comment]
@Silvenga Have you repo step fot Windows PowerShell?

**Comment by Silvenga** [comment]
@iSazonov I have an example pipeline with logs - https://github.com/Silvenga/Repo-Case-PowerShell5-Pipeline-Timeout

This issue is the same as the PowerShell Core "problem" talked about in this issue, except:
- It occurs only ~20% of the time.
- Is new behavior (used to work 100% of the time).
- Occurs on only on Windows.
- Occurs only on PowerShell 5.

It is my belief this is a bug in the Azure DevOps agent, but my support case keeps ending up in this repo (I was given this issue has evidence it's not a Azure DevOps issue 🤦). 

Frustrating to say the least.

Note that we've seen a major breaking change in Azure DevOps Pipelines, where pwsh is used instead of PowerShell (even when PowerShell is requested). This is new behavior that we've tracked to 4/14. Needless to say, a lot of our scripts are broken. So, the repo case cannot be ran anymore, without Azure DevOps creating a workaround for us. The logs will have to do.

**Comment by iSazonov** [comment]
If I understand right the following happens: Start-Process runs a "tail" process, pwsh exits and Azure engine waits "tail" process exits but it never exits.

Perhaps @PaulHigin has thoughts about the issue.

**Comment by Silvenga** [comment]
Yep, that's what appears to be occurring @iSazonov - which I've been told is a pwsh bug. I think I proved that pwsh is correctly spawning new processes under at least Linux.

Running Bash/Batch to execute PowerShell does not cause this behavior. So, this is not normal behavior for Tasks (orphan cleanup in fact occurs by the agent when the Job is complete). Again, I do not believe this is an issue with pwsh, but if Azure DevOps Product believes it is 🤷.

What's worse is if the first PowerShell Task timeouts, every PowerShell Task timeouts afterwards (a simple Write-Host will timeout). This suggests that the Azure DevOps agent isn't simply waiting for an orphaned process to stop, but some global agent state is getting messed up.

**Comment by PaulHigin** [comment]
Adding @daxian-dbw  @TravisEz13 

This sounds like a random issue when running in Azure DevOps.  There doesn't seem to be a repro in PowerShell running outside Azure DevOps.

**Comment by TravisEz13** [comment]
@Silvenga  Why to just run `tail -f /dev/nul` ?

**Comment by TravisEz13** [comment]
@Silvenga To debug the issue, we need Azure DevOps to provide us a repro that we can debug.

**Comment by Silvenga** [comment]
@TravisEz13 just for this contributed example. I needed some command that never exits (and exists by default on most Linux distributions), my first thought was `tail -f /dev/null`. This command is purely to create a simple reproduction case.

In a real-world example, we are starting a HTTP interception proxy (think Fiddler but accessible via an API) in some of our end-to-end tests, to test the L7 behavior of a third party.

In a nutshell, the example would look like:
- The interception API is started, listening for HTTP Proxy connections.
- The third party under test is started.
- We execute our testing payload against the third party.
- We assert on the observed communication from the third party through the HTTP interception proxy.
- Cleanup.

**Comment by TravisEz13** [comment]
@Silvenga So again, the azure DevOps environments are not debuggable and making the report not actionable and I go back to this.  Unfortunately, I'm going back to https://github.com/PowerShell/PowerShell/issues/11405#issuecomment-614779230

**Comment by Silvenga** [comment]
I can build out an Azure DevOps pipeline, if that would be helpful.

IMO, there are two separate problems here (that are not bugs in pwsh).
- The Azure DevOps agent is incorrectly handling/tracking processes that pwsh starts.
- The Azure DevOps agent is incorrectly handling/tracking processes that PowerShell 5 starts.

I would agree with you @TravisEz13 - I need something to go back to Azure DevOps support with.

**Comment by Silvenga** [comment]
Would you say that, given a similar issue occurs under both PowerShell 5 and pwsh, that this is not a pwsh issue and that this issue should be closed as likely a problem with the Azure DevOps agent?

**Comment by TravisEz13** [comment]
The pipeline doesn't help as it's not remotely accessible.  We would need a pipeline with a self-hosted agent that both repro's the problem and we can access to debug the issue.

**Comment by TravisEz13** [comment]
@Silvenga I'm going to mark it as answered.  This will close it after 24 hours of inactivity.

**Comment by TravisEz13** [comment]
I'll try to ask Azure DevOps if they are willing to provide us an environment, but I think it's more likely if you ask.

**Comment by Silvenga** [comment]
I'll ask, thanks for your help.

**Comment by Silvenga** [comment]
I've requested an environment from the Azure DevOps team through the open support case.

**Comment by TravisEz13** [comment]
@Silvenga I've sent an email to you to the email in your GitHub profile so you have a more private way to contact me if that is needed.
